src/SimpleRADOSStriper.cc

```diff
@@ -488,7 +488,7 @@ ssize_t SimpleRADOSStriper::read(void* data, size_t len, uint64_t off)
   }
 
   size_t r = 0;
-  std::vector<std::pair<bufferlist, aiocompletionptr>> reads;
+  std::deque<std::pair<bufferlist, aiocompletionptr>> reads;
```

Comment:
aha! Well done everyone for figuring this out.

Please add a comment as to why we don't use a vector here. Some enterprising future dev may think this can be vector instead.